front/src/modules/activities/hooks/useOpenCreateActivityDrawer.ts

```diff
@@ -37,6 +41,33 @@ export function useOpenCreateActivityDrawer() {
     entities?: ActivityTargetableEntity[],
   ) {
     const now = new Date().toISOString();
+
+    function getRelationData(
+      entities: ActivityTargetableEntity[],
+    ): ActivityTargetCreateManyActivityInput[] {
+      let relationData: ActivityTargetCreateManyActivityInput[] = [];
+      for (const entity of entities ?? []) {
+        relationData.push({
+          companyId:
+            entity.type === ActivityTargetableEntityType.Company
+              ? entity.id
+              : null,
+          personId:
+            entity.type === ActivityTargetableEntityType.Person
+              ? entity.id
+              : null,
+          id: v4(),
+          createdAt: now,
+        });
+        if (entity.relatedEntities) {
+          relationData = relationData.concat(
+            getRelationData(entity.relatedEntities),
```

Comment:
Why are we doing recursion here ? Previously it was a map no ?

Reply:
Previously an array was passed to this function (`ActivityTargetableEntity[])` but now each of this entities could itself contain an array of `ActivityTargetableEntitiy` so the thought was to allow a recursion through these arrays for future use. But if this is not needed it can be also removed and replaced by two maps :)

Reply:
The result should be the same as this recursion only influences which entities are added to the note, but for now it is  removed ...